SimGeneral/PreMixingModule/plugins/PreMixingMuonWorkers.cc

```diff
@@ -0,0 +1,218 @@
+  }
+
+  template <typename DigiCollection>
+  void PreMixingMuonWorker<DigiCollection>::addPileups(edm::EventPrincipal const& ep, edm::ModuleCallingContext const *mcc) {
+    std::shared_ptr<Wrapper<DigiCollection>  const> digisPTR =  getProductByTag<DigiCollection>(ep, pileupTag_, mcc);
+    for(const auto& elem: *(digisPTR->product())) {
+      accumulated_->put(elem.second, elem.first);
+    }
+  }
+
+  template <typename DigiCollection>
+  void PreMixingMuonWorker<DigiCollection>::put(edm::Event& iEvent) {
+    iEvent.put(std::move(accumulated_), collectionDM_);
+  }
+
+  // Specialize put for CSC strip
+  template <>
+  void PreMixingMuonWorker<CSCStripDigiCollection>::put(edm::Event& iEvent) {
+    auto merged = std::make_unique<CSCStripDigiCollection>();
+    for(const auto& elem: *accumulated_) {
+      // The layerId
+      const CSCDetId& layerId = elem.first;
+
+      // Get the iterators over the digis associated with this LayerId
+      const CSCStripDigiCollection::Range& range = elem.second;
+
+      std::vector<CSCStripDigi> NewDigiList;
+
+      std::vector<int> StripList;
+      std::vector<CSCStripDigiCollection::const_iterator> StripPointer;
+
+      for(CSCStripDigiCollection::const_iterator dtdigi=range.first; dtdigi!=range.second; ++dtdigi){
+        //std::cout << "Digi " << (*dtdigi).getStrip() << std::endl;
+	StripList.push_back( (*dtdigi).getStrip() );
+	StripPointer.push_back( dtdigi );
+      }
+
+      int PrevStrip = -1;
+      std::vector<int> DuplicateList;
+
+      std::vector<CSCStripDigiCollection::const_iterator>::const_iterator StripPtr = StripPointer.begin();
+
+      for( std::vector<int>::const_iterator istrip = StripList.begin(); istrip !=StripList.end(); ++istrip) {
+       
+	const int CurrentStrip = *(istrip);
+	
+	if(CurrentStrip > PrevStrip) { 
+	  PrevStrip = CurrentStrip; 
+
+	  int dupl_count;
+	  dupl_count = std::count(StripList.begin(), StripList.end(), CurrentStrip);
+	  if(dupl_count > 1) {
+	    std::vector<int>::const_iterator  duplicate = istrip;
+	    ++duplicate;
+	    std::vector<CSCStripDigiCollection::const_iterator>::const_iterator DuplPointer = StripPtr;
+	    ++DuplPointer;
+	    for( ; duplicate!=StripList.end(); ++duplicate) {
+	      if( (*duplicate) == CurrentStrip ) {
+
+		//		std::cout << " Duplicate of current " << CurrentStrip << " found at " << (duplicate - StripList.begin()) << std::endl;
+
+		DuplicateList.push_back(CurrentStrip);
+
+		std::vector<int> pileup_adc = (**DuplPointer).getADCCounts();
+		std::vector<int> signal_adc = (**StripPtr).getADCCounts();
+
+		std::vector<int>::const_iterator minplace;
+		
+		minplace = std::min_element(pileup_adc.begin(), pileup_adc.end());
+
+		int minvalue = (*minplace);
+		
+		std::vector<int> new_adc;
+
+		std::vector<int>::const_iterator newsig = signal_adc.begin();
+
+		for(std::vector<int>::const_iterator ibin = pileup_adc.begin(); ibin!=pileup_adc.end(); ++ibin) {
```

Comment:
It would be nice to go through all of this and modernize the code (e.g. range-based loops). Maybe in a subsequent PR...